test/e2e/service_options_test.go

```diff
@@ -167,6 +172,22 @@ func (test *e2eTest) validateServiceAnnotations(t *testing.T, r *KnRunResultColl
 	}
 }
 
+func (test *e2eTest) validateLabels(t *testing.T, r *KnRunResultCollector, serviceName string, serviceLabels, revisionLabels map[string]string) {
+	metadataLabelsJsonpathFormat := "jsonpath={.metadata.labels.%s}"
+	templateLabelsJsonpathFormat := "jsonpath={.spec.template.metadata.labels.%s}"
+
+	for k, v := range serviceLabels {
+		out := test.kn.Run("service", "describe", serviceName, "-o", fmt.Sprintf(metadataLabelsJsonpathFormat, k))
```

Comment:
Could you not run the command once and verify output with all labels after?

Reply:
Thanks for the feedback. I have updated the PR to do the same.